platform/darwin/src/MGLOpenGLStyleLayer.mm

```diff
@@ -116,9 +116,7 @@ - (void)setStyle:(MGLStyle *)style {
         [NSException raise:@"MGLLayerReuseException"
                     format:@"%@ cannot be added to more than one MGLStyle at a time.", self];
     }
-    _style.openGLLayers[self.identifier] = nil;
     _style = style;
-    _style.openGLLayers[self.identifier] = self;
```

Comment:
This is the only place where this dictionary was being used @1ec5 ; transferring the ownership is a better solution (and fixes the crash)